src/representations/representation_manager_base.hh

```diff
@@ -33,25 +33,18 @@
 
 #include <string>
 #include <vector>
+#include <iostream>
 
 namespace rascal {
 
-  enum class Option {
-    // Coulomb Matrix Options
-    CMSortDistance,
-    CMSortRowNorm,
-  };
-
-  template <class RepresentationImplementation>
-  struct RepresentationManager_traits {};
-
   class RepresentationManagerBase {
    public:
     //! type for the hyper parameter class
     using hypers_t = json;
     //! type for representation
-    // TODO(felix) Should the user have freedom for the type ?
     using precision_t = double;
+    //! type used to register the valid key and values of hypers_t
+    using reference_hypers_t = std::map<std::string, std::vector<std::string>>;
```

Comment:
Is `reference_hypers` going to be used for both hypers (generally numerical parameters) and options (switches that change the behaviour of the class)? Does the initial validation pass treat both of these on an equal footing? If so, where does the split between options and hypers happen?

Reply:
I agree with you it is tricky. I am treating them as equals here while one could imagine having option dependent hypers. The check here looks whether the field name is valid and if it as only some possible values then check if the value is valid.
The broader question that you raise is: should we check for internal consistency thoroughly ? e.g. given the representation and a set of options should we make sure that all the required hypers are present ? Is that right ?
I am not sure it is needed at this stage but if we want to go down that way we need to make the registration as simple as possible.